src/packages/migrate/src/commands/DbDrop.ts

```diff
@@ -0,0 +1,177 @@
+    const args = arg(argv, {
+      '--help': Boolean,
+      '-h': '--help',
+      '--force': Boolean,
+      '-f': '--force',
+      '--schema': String,
+      '--telemetry-information': String,
+    })
+
+    if (isError(args)) {
+      return this.help(args.message)
+    }
+
+    if (args['--help']) {
+      return this.help()
+    }
+
+    const schemaPath = await getSchemaPath(args['--schema'])
+
+    if (!schemaPath) {
+      throw new Error(
+        `Could not find a ${chalk.bold(
+          'schema.prisma',
+        )} file that is required for this command.\nYou can either provide it with ${chalk.greenBright(
+          '--schema',
+        )}, set it as \`prisma.schema\` in your package.json or put it into the default location ${chalk.greenBright(
+          './prisma/schema.prisma',
+        )} https://pris.ly/d/prisma-schema-location`,
+      )
+    }
+
+    console.info(
+      chalk.dim(
+        `Prisma Schema loaded from ${path.relative(process.cwd(), schemaPath)}`,
+      ),
+    )
+
+    const datamodel = fs.readFileSync(schemaPath, 'utf-8')
+    const config = await getConfig({ datamodel })
+    const activeDatasource = config.datasources[0]
+    const credentials = uriToCredentials(activeDatasource.url.value)
+    const { schemaWord, dbType, dbName } = getDbinfoFromCredentials(credentials)
+    const schemaDir = (await getSchemaDir(schemaPath))!
+
+    console.info() // empty line
+
+    if (!args['--force']) {
+      if (isCi) {
+        throw Error(
+          `Use the --force flag to use the drop command in an unnattended environment like ${chalk.bold.greenBright(
+            getCommandWithExecutor('prisma db drop --force'),
+          )}`,
+        )
+      }
+
+      const confirmation = await prompt({
+        type: 'text',
+        name: 'value',
+        message: `Enter the ${dbType} ${schemaWord} name "${dbName}" to drop it.\nLocation: "${getDbLocation(
+          credentials,
+        )}".\n${chalk.red('All data will be lost')}.`,
+      })
+      console.info() // empty line
+
+      if (!confirmation.value) {
+        console.info('Drop cancelled.')
+        process.exit(0)
+      } else if (confirmation.value !== dbName) {
+        throw Error(
+          `The ${schemaWord} name entered "${confirmation.value}" doesn't match "${dbName}".`,
+        )
+      }
+    }
+
+    let result: execa.ExecaReturnValue<string> | undefined = undefined
+    try {
+      result = await dropDatabase(activeDatasource.url.value, schemaDir)
+    } catch (e) {
+      let json
+      try {
+        json = JSON.parse(e.stdout)
+      } catch (e) {
+        console.error(
+          `Could not parse database drop engine response: ${e.stdout.slice(
+            0,
+            200,
+          )}`,
+        )
+      }
+
+      if (json.message) {
+        throw Error(json.message)
+      }
+
+      throw Error(e)
+    }
+
+    if (
+      result?.exitCode === 0 &&
+      result.stderr.includes('The database was successfully dropped')
```

Comment:
Just to double check, maybe you want to check for `result` to be there.
A simple `if(!result) { throw...}` can already be enough.

Reply:
But result (execa response) is always there if engines returns something without error